src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOrchestrator.hpp

```diff
@@ -262,49 +262,64 @@ class TScanOrchestrator final : public TOSPrimitives
         // - There are delayed events for that agent
         // - The event is not a cleanup or re-scan all agents event
         // Cleanup and re-scan events purge the delayed queue, so they have priority to be processed
-        if (!isDelayed && m_eventDelayedDispatcher->size(context->agentId()) &&
-            type != ScannerType::CleanupAllAgentData && type != ScannerType::ReScanAllAgents &&
-            type != ScannerType::CleanupSingleAgentData)
+        if (!isDelayed && type != ScannerType::CleanupAllAgentData && type != ScannerType::ReScanAllAgents &&
+            type != ScannerType::CleanupSingleAgentData && m_eventDelayedDispatcher->size(context->agentId()))
```

Comment:
In a previous commit, we changed the order in which we performed these operations. We were performing the following check `m_eventDelayedDispatcher->size(context->agentId())` in events without agentId

Reply:
## Testing
```
./router_test_tool -m publisher -t wdb-agent-events
Press 'q' to exit.
Failed to set socket options
Failed to set socket options
Failed to set socket options
Failed to set socket options
{"action":"cleanup","no-index":false}
Sending message: {"action":"cleanup","no-index":false}

{"action":"reboot","no-index":false}
Sending message: {"action":"reboot","no-index": false}

{"action":"deleteAgent","no-index":false,"agent_info":{"agent_id":"100"}}
Sending message: {"action":"deleteAgent","no-index":false,"agent_info":{"agent_id":"100"}}
```

```
wazuh-modulesd:content-updater:skipStep.hpp:48 handleRequest : SkipStep - Starting process
wazuh-modulesd:content-updater:cleanUpContent.hpp:63 handleRequest : CleanUpContent - Starting process
wazuh-modulesd:content-updater:action.hpp:221 runAction : Action for 'vulnerability_feed_manager' finished
wazuh-modulesd:vulnerability-scanner:vulnerabilityScannerFacade.cpp:501 start : Vulnerability scanner module started.
Press enter to stop the scanner...
Failed to set socket options
Failed to set socket options
Failed to set socket options
Failed to set socket options
wazuh-modulesd:vulnerability-scanner:scanOrchestrator.hpp:307 run : Cleanup all agents
wazuh-modulesd:vulnerability-scanner:scanOrchestrator.hpp:297 run : Re-scan all agents
Error modifying FD from interface.
wazuh-modulesd:vulnerability-scanner:scanOrchestrator.hpp:312 run : Cleanup agent 100
```